src/modules/portfolio/components/positions/positions.jsx

```diff
@@ -1,58 +1,85 @@
 import { Helmet } from 'react-helmet'
+import { Link } from 'react-router-dom'
+import makePath from 'modules/routes/helpers/make-path'
 
 import PositionsMarketsList from 'modules/portfolio/components/positions-markets-list/positions-markets-list'
 import { TYPE_CHALLENGE } from 'modules/market/constants/link-types'
+import PortfolioStyles from 'modules/portfolio/components/portfolio-view/portfolio-view.styles'
+import { MARKETS } from 'modules/routes/constants/views'
 
-const Positions = p => (
-  <section>
-    <Helmet>
-      <title>Positions</title>
-    </Helmet>
-    <PositionsMarketsList
-      title="Open Positions"
-      markets={p.openPositionMarkets}
-      closePositionStatus={p.closePositionStatus}
-      scalarShareDenomination={p.scalarShareDenomination}
-      orderCancellation={p.orderCancellation}
-      location={p.location}
-      history={p.history}
-    />
-    <PositionsMarketsList
-      title="Reporting"
-      markets={p.reportingMarkets}
-      closePositionStatus={p.closePositionStatus}
-      scalarShareDenomination={p.scalarShareDenomination}
-      orderCancellation={p.orderCancellation}
-      location={p.location}
-      history={p.history}
-      linkType={TYPE_CHALLENGE}
-      positionsDefault={false}
-    />
-    <PositionsMarketsList
-      title="Closed"
-      markets={p.closedMarkets}
-      closePositionStatus={p.closePositionStatus}
-      scalarShareDenomination={p.scalarShareDenomination}
-      orderCancellation={p.orderCancellation}
-      location={p.location}
-      history={p.history}
-      positionsDefault={false}
-    />
-  </section>
-)
+export default class Positions extends Component {
+  static propTypes = {
+    location: PropTypes.object.isRequired,
+    history: PropTypes.object.isRequired,
+    openPositionMarkets: PropTypes.array.isRequired,
+    reportingMarkets: PropTypes.array.isRequired,
+    closedMarkets: PropTypes.array.isRequired,
+    closePositionStatus: PropTypes.object.isRequired,
+    scalarShareDenomination: PropTypes.object.isRequired,
+    orderCancellation: PropTypes.object.isRequired,
+    loadAccountTrades: PropTypes.func.isRequired,
+    marketsCount: PropTypes.number.isRequired,
+    registerBlockNumber: PropTypes.number,
+  }
 
-Positions.propTypes = {
-  location: PropTypes.object.isRequired,
-  history: PropTypes.object.isRequired,
-  openPositionMarkets: PropTypes.array.isRequired,
-  reportingMarkets: PropTypes.array.isRequired,
-  closedMarkets: PropTypes.array.isRequired,
-  closePositionStatus: PropTypes.object.isRequired,
-  scalarShareDenomination: PropTypes.object.isRequired,
-  orderCancellation: PropTypes.object.isRequired,
-  registerBlockNumber: PropTypes.number
-}
+  componentWillMount() {
+    this.props.loadAccountTrades()
+  }
 
-export default Positions
+  render() {
+    const p = this.props
+    return (
+      <section>
+        <Helmet>
+          <title>Positions</title>
+        </Helmet>
+        { p.marketsCount !== 0 &&
```

Comment:
You can make this just one conditional that wraps those three components.  You’ll also need to wrap them in a div, so you might need to also update the styling a bit.